watertap/unit_models/electrodialysis_1D.py

```diff
@@ -0,0 +1,1308 @@
+            else:
+                return (
+                    self.current_efficiency_x[t, x]
+                    * self.current_density_x[t, x]
+                    * self.cell_width
+                    == -sum(
+                        self.diluate.mass_transfer_term[t, x, p, j]
+                        * self.config.property_package.charge_comp[j]
+                        for j in cation_set
+                    )
+                    * Constants.faraday_constant
+                )
+
+    # Intialization routines
+    def initialize_build(
+        blk,
+        state_args=None,
+        outlvl=idaeslog.NOTSET,
+        solver=None,
+        optarg=None,
+        fail_on_warning=False,
+        ignore_dof=False,
+    ):
+        """
+        General wrapper for pressure changer initialization routines
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                         package(s) to provide an initial state for
+                         initialization (see documentation of the specific
+                         property package) (default = {}).
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating which solver to use during
+                     initialization (default = None)
+            fail_on_warning : boolean argument to fail or only produce  warning upon unsuccessful solve (default=False)
+            ignore_dof : boolean argument to ignore when DOF != 0 (default=False)
+
+        Returns: None
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+        # Set solver options
+        opt = get_solver(solver, optarg)
+        # Set the intial conditions over the 1D length from the state vars
+        for k in blk.keys():
+            for set in blk[k].diluate.properties:
+                if (
+                    "flow_mol_phase_comp"
+                    not in blk[k].diluate.properties[set].define_state_vars()
+                ):
+                    raise ConfigurationError(
+                        "Electrodialysis1D unit model requires "
+                        "either a 'flow_mol_phase_comp' or 'flow_mass_phase_comp' "
+                        "state variable basis to apply the 'propogate_initial_state' method"
+                    )
+                if "teperature" in blk[k].diluate.properties[set].define_state_vars():
+                    blk[k].diluate.properties[set].temperature = value(
+                        blk[k].diluate.properties[(0.0, 0.0)].temperature
+                    )
+                if "pressure" in blk[k].diluate.properties[set].define_state_vars():
+                    blk[k].diluate.properties[set].pressure = value(
+                        blk[k].diluate.properties[(0.0, 0.0)].pressure
```

Comment:
For time, we have used t=0 everywhere to represent the steady state so I think it's ok to use 0 here (essentially there is no time index in our models so far) .  For length, the same is true here because the 1d control volume takes a constant continues set as (0, 1) when it was build in the current version.  So writing it as `...first()` in lieu of 0 will not make a difference if we do not give the flexibility to the control volume block's length set.    But ok to change.